app/scripts/metamask-controller.js

```diff
@@ -1986,36 +1986,28 @@ export default class MetamaskController extends EventEmitter {
   }
 
   /**
-   * A method for providing our API over a stream using Dnode.
+   * A method for providing our API over a stream using JSON-RPC.
    * @param {*} outStream - The stream to provide our API over.
    */
   setupControllerConnection(outStream) {
     const api = this.getApi();
-    // the "weak: false" option is for nodejs only (eg unit tests)
-    // it is a workaround for node v12 support
-    const dnode = Dnode(api, { weak: false });
+
     // report new active controller connection
     this.activeControllerConnections += 1;
     this.emit('controllerConnectionChanged', this.activeControllerConnections);
-    // connect dnode api to remote connection
-    pump(outStream, dnode, outStream, (err) => {
-      // report new active controller connection
-      this.activeControllerConnections -= 1;
-      this.emit(
-        'controllerConnectionChanged',
-        this.activeControllerConnections,
-      );
-      // report any error
-      if (err) {
-        log.error(err);
-      }
-    });
-    dnode.on('remote', (remote) => {
-      // push updates to popup
-      const sendUpdate = (update) => remote.sendUpdate(update);
-      this.on('update', sendUpdate);
-      // remove update listener once the connection ends
-      dnode.on('end', () => this.removeListener('update', sendUpdate));
+
+    // set up postStream transport
+    outStream.on('data', createMetaRPCHandler(api, outStream));
+    const handleUpdate = (update) => {
+      // send notification to client-side
+      outStream.write({
+        method: 'sendUpdate',
```

Comment:
Should this include `jsonrpc: '2.0'`, for spec compliance? Not that it _really_ matters since this is internal, but :shrug: 

```suggestion
        jsonrpc: '2.0',
        method: 'sendUpdate',
```

Reply:
This goes for the other messages as well - I see that it's missing for all of them.

Reply:
Yea, I don't have a preference, was hoping to get some feedback.

The current iteration: leaving it out + using params-by-position (array) gives us the least overhead. 

/cc @rekmarks

Reply:
Omitting it for internal connections seems OK to be, for the sake of saving a few bytes on each message.